fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/Comparisons.java

```diff
@@ -940,23 +968,42 @@ public ListComparison(@Nonnull Type type, @Nonnull List comparand) {
                 default:
                     throw new RecordCoreException("ListComparison only supports EQUALS, NOT_EQUALS and STARTS_WITH");
             }
-            if (comparand == null || (this.type == Type.IN && comparand.stream().anyMatch(o -> o == null))) {
-                throw new NullPointerException("List comparand is null, or contains null");
+            if (comparand == null) {
+                throw new NullPointerException("List comparand is null");
```

Comment:
I suppose I should have asked about this the first time, but do we actually want this? I don't think a Protobuf message can contain `null`, which I think is why this was written the way it was. Unless there's a reason we're changing this now?